### ISSUE TITLE
Help on training

### ISSUE THREAD

**Issue by liuxing007** [TaatiTeam/MotionAGFormer#1]
Great Job!
This is the result after 60 epochs, I'd appreciate any information you could give me.  So the epochs should be set 120 as in MotionBERT?
And I wondered whether your  dual-stream architecture can be applied in Skeleton-based action recognition?
________________________________________________________
Protocol #1 Error (MPJPE): 41.418504782180044 mm
Acceleration error: 1.2090892616283204 mm/s^2
Protocol #2 Error (P-MPJPE): 34.45834107454132 mm

**Comment by SoroushMehraban** [comment]
Are you training the base variant? Based on my experiment, base variant should reach it's best MPJPE at epoch 60.  It's not needed to train for 120 epochs.
One thing I checked now is that at the time I did the experiment, learning rate was set to 0.0005 but what I placed in config file now is 0.0002.
Please change the learning rate to see if it improves.

**Comment by liuxing007** [comment]
Thanks! After changing the learning rate, the results are consistent with the paper.

**Comment by liuxing007** [comment]
wandb: Run summary:
wandb:          eval/acceleration_error 1.0266
wandb:                   eval/min_mpjpe 39.02154
wandb:                       eval/mpjpe 39.75427
wandb:                     eval/p-mpjpe 33.19672
wandb:       eval_additional/1_DF_error 32.24598
wandb:       eval_additional/2_DF_error 42.13039
wandb:       eval_additional/3_DF_error 59.86381
wandb: eval_additional/lower_body_error 38.25731
wandb: eval_additional/upper_body_error 44.62787
wandb:         eval_joints/Bottom torso 0.0
wandb:          eval_joints/Center head 38.25136
wandb:         eval_joints/Center torso 27.47751
wandb:           eval_joints/Left elbow 53.68705
wandb:            eval_joints/Left foot 57.07191
wandb:            eval_joints/Left hand 62.90114
wandb:             eval_joints/Left hip 21.00947
wandb:            eval_joints/Left knee 35.37488
wandb:        eval_joints/Left shoulder 41.57694
wandb:            eval_joints/Neck base 39.00301
wandb:          eval_joints/Right elbow 47.91908
wandb:           eval_joints/Right foot 60.23595
wandb:           eval_joints/Right hand 59.24625
wandb:            eval_joints/Right hip 20.43206
wandb:           eval_joints/Right knee 35.41957
wandb:       eval_joints/Right shoulder 39.20841
wandb:          eval_joints/Upper torso 37.00795
wandb:                               lr 0.00028
wandb:             train/angle_velocity 0.00429
wandb:               train/loss_2d_proj 0
wandb:               train/loss_3d_pose 0.00603
wandb:              train/loss_3d_scale 0.00593
wandb:           train/loss_3d_velocity 0.0006
wandb:                 train/loss_angle 0.04273
wandb:                    train/loss_lg 0.00167
wandb:                    train/loss_lv 9e-05
wandb:                      train/total 0.02091
___________________________________________________________
Thanks for your response. After  setting LR to 0.0005, the results of training the base variant is shown as above. Best results is 39.02mm.

**Comment by SoroushMehraban** [comment]
I've noticed that achieving the exact same numbers is not possible in our lab, where we have V100, A40, and RTX 3090 GPUs. One interesting observation from my experiments is that when I switched from the A40 to the RTX, the MPJPE I achieved was different. Similarly, even when I changed from 2 A40 GPUs to 4 GPUs, the results differed. I even had the same issue with MotionBERT and couldn't replicate their result, which I assume is because of having different type of GPUs. But I have provided the checkpoints which you can load and verify.

**Comment by zerowing-ex** [comment]
I think the main reason why the results differ for different GPUs is the batch size. The A40 GPU owns 48GB of VRAM, while 24GB and 32GB are for RTX 3090/4090 and V100, respectively. The VRAM limits the batch size; therefore, assuming that you're using RTX 4090 GPUs and the batch size is 4 (takes ~20GB of VRAM), but the batch size was originally 16 on A40. If you want to reproduce the result of the paper, you should use the grad accumulation techniques, i.e., apply torch.zero_grad() for each step (since 4*4=16), and I'd like to check this later.

Environments:
```
OS: Ubuntu 23.04
GPU: 4*RTX4090
torch: 2.1.0_py3.8_cuda12.1_cudnn8.9.2_0
CUDA: 12.1
cuDNN: 8.9.2

NOTE: the experiments as follows are **only use 1 RTX4090**
```

Experiment 1:
```
lr=0.0002
batch_size=4
```

Best Result (epoch 34):
```
Protocol #1 Error (MPJPE): 39.0806232811987 mm
Acceleration error: 1.1593467465534728 mm/s^2
Protocol #2 Error (P-MPJPE): 32.898190088240604 mm
```

Experiment 2:
```
lr=0.0005
batch_size=4
```

Best Result (epoch 26):
```
Protocol #1 Error (MPJPE): 38.69955514154377 mm
Acceleration error: 1.004794544925152 mm/s^2 
Protocol #2 Error (P-MPJPE): 32.321134679653895 mm
```
